src/webview/components/memory-table.tsx

```diff
@@ -182,6 +188,11 @@ export class MemoryTable extends React.PureComponent<MemoryTableProps, MemoryTab
 
         return (
             <div className='flex-1 overflow-auto px-4'>
+                <Tooltip
+                    onBeforeShow={this.handleOnBeforeTooltipShow}
```

Comment:
This seems a bit odd to me: We need to trigger a callback before the tooltip is shown which triggers the update of the tooltip content by setting the state in the component. I assume there really isn't any other way or maybe I misunderstood but I think if you were to add a small comment about this somewhere to make it more obvious how this works, that might ease the understanding.

Reply:
You're right that it is a peculiar flow. The chief advantage of doing it this way is that it lets the Tooltip self-contain all its state & handlers. I'll add the comment you suggested.

In an earlier draft of this, I tried adding event handlers to other locations, such as the DataTable as a whole, tracking which elements are mouseover'd at any time, and then giving the Tooltip access to that state. It seemed a little heavy-handed, and might have had a slight performance impact. But functionally it was fine and the flow was conceptually clearer. Let me know if you'd prefer to go back to that flow.

Reply:
I think we can stick to this flow for now, especially now that we have a comment explaining it a bit. We can always re-work it if we actually run into any kind of problems, thank you!